dozer-ingestion/src/connectors/dozer/connector.rs

```diff
@@ -0,0 +1,235 @@
+        let _ = Self::get_client(&self.config).await?;
+
+        Ok(())
+    }
+
+    async fn list_tables(&self) -> Result<Vec<TableIdentifier>, ConnectorError> {
+        let mut tables = vec![];
+        let response = self.describe_application().await?;
+        for (endpoint, _) in response.endpoints {
+            tables.push(TableIdentifier::new(None, endpoint));
+        }
+
+        Ok(tables)
+    }
+
+    async fn validate_tables(&self, _tables: &[TableIdentifier]) -> Result<(), ConnectorError> {
+        self.validate_connection().await?;
+
+        Ok(())
+    }
+
+    async fn list_columns(
+        &self,
+        _tables: Vec<TableIdentifier>,
+    ) -> Result<Vec<TableInfo>, ConnectorError> {
+        let mut tables = vec![];
+        let response = self.describe_application().await?;
+        for (endpoint, build) in response.endpoints {
+            let schema: SourceSchema = serde_json::from_str(&build.schema_string).map_err(|e| {
+                ConnectorError::TypeError(
+                    dozer_types::errors::types::TypeError::DeserializationError(
+                        DeserializationError::Json(e),
+                    ),
+                )
+            })?;
+            tables.push(TableInfo {
+                schema: None,
+                name: endpoint,
+                column_names: schema
+                    .schema
+                    .fields
+                    .iter()
+                    .map(|field| field.name.clone())
+                    .collect(),
+            });
+        }
+        Ok(tables)
+    }
+
+    async fn get_schemas(
+        &self,
+        table_infos: &[TableInfo],
+    ) -> Result<Vec<SourceSchemaResult>, ConnectorError> {
+        let mut schemas = vec![];
+        for table_info in table_infos {
+            let log_reader = self.get_reader_builder(table_info.name.clone()).await;
+
+            match log_reader {
+                Ok(log_reader) => {
+                    let mut schema = log_reader.schema.schema.clone();
+                    for field in schema.fields.iter_mut() {
+                        // All fields are dynamically defined by the upstream instance
+                        field.source = SourceDefinition::Dynamic;
+                    }
+                    schemas.push(Ok(SourceSchema::new(schema, CdcType::FullChanges)));
+                }
+                Err(e) => {
+                    schemas.push(Err(e));
+                }
+            }
+        }
+
+        Ok(schemas)
+    }
+
+    async fn start(
+        &self,
+        ingestor: &Ingestor,
+        tables: Vec<TableInfo>,
+    ) -> Result<(), ConnectorError> {
+        let mut joinset = JoinSet::new();
+        for (table_index, table) in tables.into_iter().enumerate() {
+            let builder = self.get_reader_builder(table.name.clone()).await?;
+            joinset.spawn(read_table(table_index, builder, ingestor.clone()));
```

Comment:
We'll need to make the ingestion order deterministic because we want dozer connector to support restart. Leaving a comment here to remind myself.